### ISSUE TITLE
Option to set something's value to a variable.

### ISSUE THREAD

**Issue by jayare5** [4ian/GDevelop#6941]
I know. This sounds like a feature that is as old as time itself, but yet we lost it somehow.

At some point, we lost the option to change the value of something into the value of a variable. Now, we have to remember the variable's name to be able to use its value in this particular menu. I humbly request that we have this option back, as it was much easier to use. My project has a lot of variables, and they're all named and organized in a way that it's when I look at them that I know exactly what it is.
Though typing does give a list, typing implies I already know exactly what I'm looking for, but that's way too challenging.

Thanks!

Old version:
![image](https://github.com/user-attachments/assets/2538cd3d-64a0-4c06-baa0-61ee2a89679b)

New version:
![image](https://github.com/user-attachments/assets/239c00f8-6a3c-4b1c-832e-12c4067e93bf)

**Comment by Silver-Streak** [comment]
Your submission title doesn't quite match your post.

You can still set something to the value of a variable. It is just not an expression anymore so it doesn't show up in the expression builder. As you mentioned it expects you to know your variables and just type the name.

Do you mean to request a GUI variable picker? If so I can update the title of the post

**Comment by jayare5** [comment]
> Your submission title doesn't quite match your post.
> 
> You can still set something to the value of a variable. It is just not an expression anymore so it doesn't show up in the expression builder. As you mentioned it expects you to know your variables and just type the name.
> 
> Do you mean to request a GUI variable picker? If so I can update the title of the post

Yes that's what I mean! You're right, the option exists in the form of typing it, but in my head I saw it as adding the option to the list of actions :D  you can update title for clarity.

You know, even sillier is the fact that the variables STILL show up in the variable expressions, if you force it to choose it as an "array" to count the number of children... And then I just simply delete the part that's about number of children but keep the name of the variable, even inside a parenthesis, it will work. So I think it's silly that we can do things that wrongly but not just select the variable name directly as it used to be.

**Comment by Silver-Streak** [comment]
Thanks for the clarification, updated the title.

For your other comment. Keep in mind the things still listed in the expression builder are actually expressions (array count, number of children, etc, are all still expressions).

**Comment by jayare5** [comment]
> Thanks for the clarification, updated the title.
> 
> For your other comment. Keep in mind the things still listed in the expression builder are actually expressions (array count, number of children, etc, are all still expressions).

Of course! I am in favor of being able to use the number of variable children, or the first/last number. I'm just baffled I can use that to find the regular variables but no dedicated option for it.
Why wouldn't a regular variable be part of an expression?

**Comment by Lumios64** [comment]
Any plan to solve this issue?

I not only agree with [jayare5](https://github.com/jayare5), but on top of that disallowing declaring variables as being global or belonging to a scene via an expression is a problem if I reuse the same variable names. For example let's say I may save a score as a scene variable to later add it to a global score like so `GlobalVariable(foo) + Variable(foo) = global foo`
The fact the current system cannot distinguish between scene and global variables at all bothers me, I think that should either be brought back or some pseudo expression should be used in its place like what's already being used for object variables.
